File: [1] Two Sum/1.two-sum.cpp
/*
 * @lc app=leetcode id=1 lang=cpp
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> mymap;
        for (int i=0; i<nums.size(); i++) {
            auto it = mymap.find(nums[i]);
            if (it != mymap.end()) {
                return vector<int> {i, mymap.at(nums[i])};
            }
            mymap[target-nums[i]] = i;
        }
        return vector<int>{};
    }
};
// @lc code=end

File: [1] Two Sum/1.two-sum.java

/*
 * @lc app=leetcode id=1 lang=java
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hm = new HashMap<>();
        for (int i=0; i<nums.length; i++) {
            if (hm.containsKey(nums[i])) {
                return new int[]{i, hm.get(nums[i])};
            }
            hm.put(target-nums[i], i);
        }
        throw new IllegalArgumentException("No solution!");
    }
}
// @lc code=end

